web/cobrands/fixmystreet/fixmystreet.js

```diff
@@ -179,6 +179,40 @@ fixmystreet.resize_to = {
   }
 };
 
+fixmystreet.geolocate = {
+    setup: function(success_callback) {
+        $('#geolocate_link').click(function(e) {
+            var $link = $(this);
+            e.preventDefault();
+            // Spinny thing!
+            if($('.mobile').length){
+                $link.append(' <img src="/cobrands/fixmystreet/images/spinner-black.gif" alt="" align="bottom">');
+            }else{
+                var spincolor = $('<span>').css("color","white").css("color") === $('#front-main').css("background-color")? 'white' : 'yellow';
+                $link.append(' <img src="/cobrands/fixmystreet/images/spinner-' + spincolor + '.gif" alt="" align="bottom">');
+            }
+            geo_position_js.getCurrentPosition(function(pos) {
+                $link.find('img').remove();
+                success_callback(pos);
+            }, function(err) {
+                $link.find('img').remove();
+                if (err.code == 1) { // User said no
+                    $link.html(translation_strings.geolocation_declined);
```

Comment:
When `#geolocate_link` is used on the Inspector page, this error message ("You declined; please fill in the box above") doesn't really make sense any more, since there’s no box to fill in. Do we need to make the error message configurable?

![screen shot 2016-11-03 at 16 15 22](https://cloud.githubusercontent.com/assets/739624/19974794/948a89aa-a1e1-11e6-947d-34a0721d9656.png)